src/synthesizer/particle/stars.py

```diff
@@ -798,6 +798,7 @@ def generate_particle_lnu(
         grid,
         spectra_name,
         fesc=0.0,
+        mask=None,
```

Comment:
We need to commit to one or the other I think. Adding both is unnecessarily introducing complexity.

However, which is preferable I am not convinced of.

Reply:
That being said. We should not remove the old way right now and should raise a deprecation warning.